src/Symfony/Component/Security/Core/Authorization/ExpressionLanguageProvider.php

```diff
@@ -25,8 +25,12 @@ public function getFunctions()
     {
         return [
             new ExpressionFunction('is_anonymous', function () {
+                @trigger_error("is_anonymous() is deprecated since version 5.1 and will be removed in 6.0. Use is_granted('IS_ANONYMOUS') instead.", E_USER_DEPRECATED);
```

Comment:
Should use the new trigger_deprecation()
But: do we really need to deprecate those?
Why requiring ppl to change their code while we can keep it working?

Reply:
Agreed, let's be more conservative. I've reverted this commit as well